Crash on empty documents

Hello,
I noticed a crash when running on documents with no titles:
```
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                     
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0xa17222]                                                                                                                                     
                                                                                                                                                                                                            
goroutine 1 [running]:                                                                                                                                                                                      
github.com/yuin/goldmark/renderer.(*renderer).Render.func2({0x0, 0x0}, 0x40?)                                                                                                                               
        /home/cdupont/go/pkg/mod/github.com/yuin/goldmark@v1.5.5/renderer/renderer.go:164 +0x62                                                                                                             
github.com/yuin/goldmark/ast.walkHelper({0x0, 0x0}, 0xc00073f930)                                                                                                                                           
        /home/cdupont/go/pkg/mod/github.com/yuin/goldmark@v1.5.5/ast/ast.go:492 +0x34                                                                                                                       
github.com/yuin/goldmark/ast.Walk(...)                                                                                                                                                                      
        /home/cdupont/go/pkg/mod/github.com/yuin/goldmark@v1.5.5/ast/ast.go:487                                                                                                                             
github.com/yuin/goldmark/renderer.(*renderer).Render(0xc00020ee10?, {0xf45780?, 0xc00012e008?}, {0xc0005ad4e5?, 0x5?, 0x5?}, {0x0?, 0x0?})                                                                  
        /home/cdupont/go/pkg/mod/github.com/yuin/goldmark@v1.5.5/renderer/renderer.go:161 +0x225                                                                                                            
github.com/Waziup/SolutionLabServer/pkg/lab.parseHugoDocument(0x1, {0xc00018a600, 0x72}, {0xc000094200, 0x21, 0x200})                                                                                       
        /home/cdupont/Documents/WaziupEV/LearningCenter/learning-center-backend/pkg/lab/import.go:972 +0xa48                                                                                                
github.com/Waziup/SolutionLabServer/pkg/lab.readHugoDocument(0xc00073fd50?, {0xc000264310, 0x68}, 0x1?)                                                                                                     
        /home/cdupont/Documents/WaziupEV/LearningCenter/learning-center-backend/pkg/lab/import.go:790 +0x258                                                                                                
github.com/Waziup/SolutionLabServer/pkg/lab.readHugoModules(0x1, {0xc0000386c0, 0x5a})                                                                                                                      
        /home/cdupont/Documents/WaziupEV/LearningCenter/learning-center-backend/pkg/lab/import.go:89 +0x285                                                                                                 
github.com/Waziup/SolutionLabServer/pkg/lab.Import(0xc000130120?, {0x7ffd9da2a18a, 0x53})                                                                                                                   
        /home/cdupont/Documents/WaziupEV/LearningCenter/learning-center-backend/pkg/lab/import.go:44 +0x179                                                                                                 
main.main()                                                                                                       
```

Example:
```
func Example() {
	src := []byte(`
Hello
`)

	markdown := goldmark.New()

	markdown.Parser().AddOptions(parser.WithAutoHeadingID())

	doc := markdown.Parser().Parse(text.NewReader(src))

	tree, err := toc.Inspect(doc, src)
	if err != nil {
		panic(err)
	}

	treeList := toc.RenderList(tree)

	if err := markdown.Renderer().Render(os.Stdout, src, treeList); err != nil {
		panic(err)
	}

}
```

--- a/README.md
+++ b/README.md
@@ -179,8 +179,15 @@ You may manipulate the `tree` before rendering the list.
 Finally, render this table of contents along with your Markdown document:
 
 ```go
-markdown.Renderer().Render(output, src, list) // table of contents
-markdown.Renderer().Render(output, src, doc)  // document
+// Render the table of contents.
+if list != nil {
+    // list will be nil if the table of contents is empty
+    // because there were no headings in the document.
+    markdown.Renderer().Render(output, src, list)
+}
+
+// Render the document.
+markdown.Renderer().Render(output, src, doc)
 ```
 
 Alternatively, include the table of contents into your Markdown document in
@@ -188,7 +195,9 @@ your desired position and render it using your Markdown renderer.
 
 ```go
 // Prepend table of contents to the front of the document.
-doc.InsertBefore(doc, doc.FirstChild(), list)
+if list != nil {
+    doc.InsertBefore(doc, doc.FirstChild(), list)
+}
 
 // Render the document.
 markdown.Renderer().Render(output, src, doc)

--- a/doc.go
+++ b/doc.go
@@ -16,6 +16,10 @@
 // manipulate the TOC, removing items from it or simplifying it, before
 // rendering.
 //
+//	if len(tocTree.Items) == 0 {
+//		// No headings in the document.
+//		return
+//	}
 //	tocList := toc.RenderList(tocTree)
 //
 // You can render that Markdown document using goldmark into whatever form you

--- a/example_test.go
+++ b/example_test.go
@@ -43,6 +43,11 @@ Bye
 		panic(err)
 	}
 
+	if len(tree.Items) == 0 {
+		return
+		// No table of contents because there are no headers.
+	}
+
 	// Render the tree as-is into a Markdown list.
 	treeList := toc.RenderList(tree)
 

--- a/inspect.go
+++ b/inspect.go
@@ -68,6 +68,7 @@ func (d maxDepthOption) String() string {
 //
 // The table of contents is represents as a tree where each item represents a
 // heading or a heading level with zero or more children.
+// The returned TOC will be empty if there are no headings in the document.
 //
 // For example,
 //

--- a/render.go
+++ b/render.go
@@ -6,6 +6,9 @@ const _defaultMarker = '*'
 
 // RenderList renders a table of contents as a nested list with a sane,
 // default configuration for the ListRenderer.
+//
+// If the TOC is nil or empty, nil is returned.
+// Do not call Goldmark's renderer if the returned node is nil.
 func RenderList(toc *TOC) ast.Node {
 	return new(ListRenderer).Render(toc)
 }
@@ -33,7 +36,13 @@ type ListRenderer struct {
 }
 
 // Render renders the table of contents into Markdown.
+//
+// If the TOC is nil or empty, nil is returned.
+// Do not call Goldmark's renderer if the returned node is nil.
 func (r *ListRenderer) Render(toc *TOC) ast.Node {
+	if toc == nil {
+		return nil
+	}
 	return r.renderItems(toc.Items)
 }
 

--- a/render_test.go
+++ b/render_test.go
@@ -237,3 +237,9 @@ func TestRenderList(t *testing.T) {
 		})
 	}
 }
+
+func TestRenderList_nil(t *testing.T) {
+	t.Parallel()
+
+	assert.Nil(t, RenderList(nil))
+}

--- a/toc.go
+++ b/toc.go
@@ -4,6 +4,8 @@ package toc
 // rest of the table of contents resides.
 type TOC struct {
 	// Items holds the top-level headings under the table of contents.
+	//
+	// Items is empty if there are no headings in the document.
 	Items Items
 }
 

